plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -102,6 +109,16 @@ func (a *podSecurityPolicyPlugin) SetInternalKubeInformerFactory(f informers.Sha
 	a.SetReadyFunc(podSecurityPolicyInformer.Informer().HasSynced)
 }
 
+func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) error {
+	return c.metrics.Admit(a)
+}
+
+// podSecurityPolicyPluginInner provides the inner implementation of Admit so that the metrics
+// wrapper can be internal to the PodSecurityPolicy admission plugin.
+type podSecurityPolicyPluginInner struct {
```

Comment:
isn't this the outer plugin?

Reply:
It's a circular dependency, but doing it this way meant a much smaller code delta. It's called "Inner" since it get's called as the inner Admit (i.e. PodSecurityPolicyPlugin.Admit -> metrics.Admit -> podSecurityPolicyPluginInner.Admit)